pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -305,7 +306,7 @@ def gpus_metrics(self) -> Dict[str, float]:
         .. deprecated:: v1.5
             Will be removed in v1.7.
         """
-        if self.trainer._device_type == _AcceleratorType.GPU and self.log_gpu_memory:
+        if isinstance(self.trainer.accelerator, GPUAccelerator) and self.log_gpu_memory:
```

Comment:
Alternatively we can use the strategy's root device type

Reply:
I would prefer to stick with the Accelerators, as this one really is what defines the accelerator being used. For some accelerators it might make sense to have the `root_device` as cpu and then distribute it later within the strategy or something similar.